llms/mlx_lm/tuner/trainer.py

```diff
@@ -143,17 +143,15 @@ def train(
 ):
     print(f"Starting training..., iters: {args.iters}")
 
-    def check_checkpoints_path(adapter_file):
-        checkpoints_path = "checkpoints"
-        if os.path.dirname(adapter_file):
-            checkpoints_path = os.path.join(
-                os.path.dirname(adapter_file), "checkpoints"
-            )
+    def check_checkpoints_path(adapter_file) -> str:
+        checkpoints_path = Path("checkpoints")
+        if Path(adapter_file).parent:
+            checkpoints_path = Path(adapter_file).parent / "checkpoints"
 
-        if not os.path.exists(checkpoints_path):
-            os.makedirs(checkpoints_path)
+        if not checkpoints_path.exists():
+            checkpoints_path.mkdir(parents=True)
 
-        return checkpoints_path
+        return str(checkpoints_path)
 
     # Create checkpoints directory if it does not exist
     adapter_path = check_checkpoints_path(args.adapter_file)
```

Comment:
It might simplify things a bit if you 
1. Make `args.adapter_file` a `Path` here instead.
2. Make a second variable for the `checkpoint_path` (it's a bit more clear as well IMO)